aviary/examples/external_subsystems/engine_NPSS/NPSS_Model/DesignEngineGroup.py

```diff
@@ -0,0 +1,167 @@
+        Wf_training_data_snip = [
+            a for a in Wf_training_data_raw if a.replace('.', '').isnumeric()]
+        Wf_training_data = np.array(Wf_training_data_snip, float)
+        outputs['Wf_training_data'] = Wf_training_data
+
+        parser.mark_anchor("thrust_training_data")
+        parser.set_delimiters(', ')
+        thrust_training_data_raw = parser.transfer_array(0, 4, 0, vec_size*3)
+        thrust_training_data_snip = [
+            a for a in thrust_training_data_raw if a.replace('.', '').isnumeric()]
+        thrust_training_data = np.array(thrust_training_data_snip, float)
+        outputs['thrust_training_data'] = thrust_training_data
+
+        parser.mark_anchor("thrustmax_training_data")
+        parser.set_delimiters(', ')
+        thrustmax_training_data_raw = parser.transfer_array(0, 4, 0, vec_size*3)
+        thrustmax_training_data_snip = [
+            a for a in thrustmax_training_data_raw if a.replace('.', '').isnumeric()]
+        thrustmax_training_data = np.array(thrustmax_training_data_snip, float)
+        outputs['thrustmax_training_data'] = thrustmax_training_data
+        # print(thrustmax_training_data)
+
+
+class DesignEngineGroup(om.Group):
+    def setup(self):
+        self.add_subsystem('DESIGN', NPSSExternalCodeComp(),
+                           promotes_inputs=[('W_DES', Aircraft.Engine.DESIGN_MASS_FLOW)],
+                           # promotes_outputs=[('Fn_SLS', Aircraft.Engine.SCALED_SLS_THRUST), ('thrust_training_data', Dynamic.Mission.THRUST+'_train'),
+                           #                   ('thrustmax_training_data', Dynamic.Mission.THRUST_MAX+'_train'), ('Wf_training_data', 'Wf_td')])
+                           promotes_outputs=[('Fn_SLS', Aircraft.Engine.SCALED_SLS_THRUST), ('thrust_training_data', 'Fn_train'),
+                                             ('thrustmax_training_data', 'Fn_max_train'), ('Wf_training_data', 'Wf_td')])
+
+        self.add_subsystem('neg_Wf', om.ExecComp('y=-x',
+                                                 x={'val': np.ones(
+                                                     vec_size), 'units': 'lbm/s'},
+                                                 y={'val': np.ones(vec_size), 'units': 'lbm/s'}),
+                           promotes_inputs=[('x', 'Wf_td')],
+                           promotes_outputs=[('y', 'Wf_inv_train')])
+
+    def configure(self):
+        self.set_input_defaults(Aircraft.Engine.DESIGN_MASS_FLOW, 240.0, units='lbm/s')
+        self.set_input_defaults('DESIGN.Alt_DES', 0.0, units='ft')
+        self.set_input_defaults('DESIGN.MN_DES', 0.0)
+        super().configure()
+
+
+if __name__ == "__main__":
+    import openmdao.api as om
+
+    SNOPT_EN = False
+    prob = om.Problem()
+    model = prob.model
+
+    # add system model
+    model.add_subsystem('p', DesignEngineGroup())
+
+    # Add solver
+    newton = model.nonlinear_solver = om.NewtonSolver()
```

Comment:
I am not sure why we have a Newton solver here, since (as far as i can tell) the NPSS component is just and explicit component, so no cycles and no implicit states to solve.

Reply:
I removed the if main run test. uploaded

Reply:
Added a bench test